sdk/storage/Azure.Storage.Blobs/src/BlockBlobClient.cs

```diff
@@ -1293,7 +1293,7 @@ protected sealed override BlobBaseClient WithSnapshotCore(string snapshot)
                     $"{nameof(conditions)}: {conditions}");
                 try
                 {
-                    var blocks = new BlockLookupList() { Uncommitted = base64BlockIds.ToList() };
+                    var blocks = new BlockLookupList() { Latest = base64BlockIds.ToList() };
```

Comment:
We should add a test to verify this doesn't regress in the future.

Reply:
added.